File: examples/mt_testing/run/run_mt_test.rb
#!/usr/bin/env ruby

require 'anlnext' # ANL Next library
require 'myPackageMT'# Ruby extension library using ANL Next

# Define your own application class derived from ANL::ANLApp.
class MyApp < ANL::ANLApp
  # Define an analysis chain in setup() method.
  def setup()
    add_namespace MyPackageMT

    chain :MyMTModule
    with_parameters(quit_index: 500,
                    quit_all: true)

    # If you need to add the same type of module, you should set another name
    # via the second argument.
    chain :MyMTModule, :MyMTModule2
    with_parameters(quit_index: 1000,
                    quit_all: false)

  end
end

a = MyApp.new
a.num_parallels = 4

a.modify do |anl|
  anl.modify_parameters 2, :MyMTModule2, {
    quit_index: 100
  }
end

a.run(10000, 1)

File: examples/mt_testing/run_py/run_mt_test.py
#!/usr/bin/env python3

import anlnext # ANL Next library
import myPackageMTPy as myp # Python extension library using ANL Next

### chain definition
def setup(module):
    module.chain(myp.MyMTModule)
    module.with_parameters({
        "quit_index": -1,
        "quit_all": True
    })

    module.chain(myp.MyMTModule, "MyMTModule2")
    module.with_parameters({
        "quit_index": 77777,
        "quit_all": True
    })

### run analysis chain
a = anlnext.AnalysisChain()
a.num_parallels = 4
setup(a)

def modify_param(module):
    module.get_parallel_module(2, "MyMTModule2").set_parameter("quit_index", 500)

a.modify(modify_param)

a.run(1000000)
